go/vt/wrangler/reparent.go

```diff
@@ -1043,6 +1089,52 @@ func (wr *Wrangler) emergencyReparentShardLocked(ctx context.Context, ev *events
 	return nil
 }
 
+// CheckTabletIsFarthestAhead will take a tablet alias string, along with a statusMap of tablet alias strings to tablet Status objects,
+// and return an error if the tablet is not the farthest ahead, or otherwise return nil if it is the farthest ahead.
+func (wr *Wrangler) CheckTabletIsFarthestAhead(tabletAliasStr string, statusMap map[string]*replicationdatapb.Status) error {
```

Comment:
I think we actually need to ensure that the candidate master's _executed_ position is at least as advanced as every replica's relay log pos. Promoting a replica to master purges its relay log, so only the transactions that have been executed should count.